docs/1.getting-started/11.testing.md

```diff
@@ -5,36 +5,378 @@ navigation.icon: i-ph-check-circle-duotone
 If you are a module author, you can find more specific information in the [Module Author's guide](/docs/guide/going-further/modules#testing)
 ::
 
-In Nuxt 3, we have a rewritten version of `@nuxt/test-utils`. We support [Vitest](https://github.com/vitest-dev/vitest) and [Jest](https://jestjs.io) as test runners.
+Nuxt offers first-class support for e2e and unit testing of your Nuxt applications, via `@nuxt/test-utils`, a library of test utilities and configuration that currently powers [the tests we use on Nuxt itself](https://github.com/nuxt/nuxt/tree/main/test), as well as tests throughout the module ecosystem.
 
 ## Installation
 
+In order to allow you to manage your other testing dependencies, `@nuxt/test-utils` ships with various optional peer dependencies. For example:
+
+- you can choose between `happy-dom` and `jsdom` for a runtime Nuxt environment
+- you can choose between `vitest` and `jest` for e2e test runners
+- `playwright-core` is only required if you wish to use the built-in browser testing utilities
+
 ::code-group
 ```bash [yarn]
-yarn add --dev @nuxt/test-utils vitest
+yarn add --dev @nuxt/test-utils vitest @vue/test-utils happy-dom playwright-core
 ```
 ```bash [npm]
-npm i --save-dev @nuxt/test-utils vitest
+npm i --save-dev @nuxt/test-utils vitest @vue/test-utils happy-dom playwright-core
 ```
 ```bash [pnpm]
-pnpm add -D @nuxt/test-utils vitest
+pnpm add -D @nuxt/test-utils vitest @vue/test-utils happy-dom playwright-core
 ```
 ```bash [bun]
-bun add --dev @nuxt/test-utils vitest
+bun add --dev @nuxt/test-utils vitest @vue/test-utils happy-dom playwright-core
 ```
 ::
 
-## Setup
+## Unit Testing
+
+We currently ship an environment for unit testing code that needs a [Nuxt](https://nuxt.com) runtime environment. It currently _only has support for `vitest`_ (although contribution to add other runtimes would be welcome).
+
+### Setup
+
+1. Add `@nuxt/test-utils/module` to your `nuxt.config` file. This is optional, and adds a vitest integration to your Nuxt DevTools which supports running your unit tests whilst in development.
+
+   ```js
+   export default defineNuxtConfig({
+     modules: [
+       '@nuxt/test-utils/module'
+     ]
+   })
+   ```
+
+2. Create a `vitest.config.ts` with the following content:
+
+   ```js
+   import { defineVitestConfig } from '@nuxt/test-utils/config'
+   
+   export default defineVitestConfig({
+     // any custom vitest config you require
+   })
+   ```
+
+### Using a Nuxt Runtime Environment
+
+By default, `@nuxt/test-utils` will not change your default Vitest environment, so you can do fine-grained opt-in and run Nuxt tests together with other unit tests.
+
+We provided a filename convention that test files contains `.nuxt.`, like `*.nuxt.test.{js,ts}` and `*.nuxt.spec.{js,ts}`, will be run in Nuxt environment automatically.
+
+Or you can add `@vitest-environment nuxt` in your test file as a comment to opt-in per test file.
```

Comment:
```suggestion
We provide a filename convention for test files containing `.nuxt.` (e.g. `*.nuxt.test.{js,ts}` or `*.nuxt.spec.{js,ts}`), which will be run in Nuxt environment automatically. You can also add `@vitest-environment nuxt` as a comment in your test file to opt-in per test file instead.
```

Reply:
I think these are separate ways of enabling the environment and should be kept separate.

Reply:
Added _instead_ at the end of the last sentence, but yes, I agree! Numbered/bulleted list will better highlight the fact there are two options to achieve the same goal. I'd personally go for bullet points as these are equally viable options which don't need ordering.

Reply:
Some ideas (single sentence actually sounds equally good):
```suggestion
You can automatically run given test file in a Nuxt environment by adding:
- `.nuxt.` to its filename (e.g. `*.nuxt.test.{js,ts}` or `*.nuxt.spec.{js,ts}`);
- `@vitest-environment nuxt` as a comment directly in the test file.
```

```suggestion
You can automatically run given test file in a Nuxt environment by either adding `.nuxt.` to its filename (e.g. `*.nuxt.test.{js,ts}` or `*.nuxt.spec.{js,ts}`) or `@vitest-environment nuxt` as a comment directly in the test file.
```